crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -568,14 +573,18 @@ impl VM {
         );
 
         self.accrued_substate.created_accounts.insert(new_address); // Mostly for SELFDESTRUCT during initcode.
-        self.accrued_substate.touched_accounts.insert(new_address);
 
         let tx_report = self.execute(&mut new_call_frame)?;
-
+        // Return reserved gas
+        current_call_frame.gas_used = current_call_frame
+            .gas_used
+            .checked_sub(max_message_call_gas.into())
+            .ok_or(VMError::Internal(InternalError::GasOverflow))?;
+        // Consume actual gas
```

Comment:
What do you think if we change both these operations of consuming and returning for something like:

```rust
        let unused_gas = max_message_call_gas
            .checked_sub(tx_report.gas_used)
            .ok_or(InternalError::GasOverflow)?;

        // Return reserved gas
        current_call_frame.gas_used = current_call_frame
            .gas_used
            .checked_sub(unused_gas.into())
            .ok_or(InternalError::GasOverflow)?;
```
I feel this code is more "descriptive", let me know what you think
Also, you can make use of the `thiserror` crate we are using and return directly an `InternalError` instead of `VMError`. It just looks nicer but it is not relevant.

Reply:
Done in [61ec86b](https://github.com/lambdaclass/ethrex/pull/1532/commits/61ec86be9730464b634f61d0b919442797aa5bab)